src/transport/mqtt/connection.rs

```diff
@@ -488,26 +531,58 @@ impl Handshake {
         session_data: SessionData,
     ) -> Next
     where
-        S: PropertyStore,
+        S: PropertyStore + StoreCapabilities,
     {
         let client = conn.client.clone();
+        let client_id: ClientId = client_id.into();
+        let store = store.clone();
 
-        let client_id_cl: ClientId = client_id.into();
+        let handle = if self.session_present && conn.session_sync {
+            info!(
+                "Already synchronized skipping handshake (introspection synchronized: {})",
```

Comment:
i would simplify in something like 

```suggestion
                "Introspection already synchronized: {}",
```